File: task2.js
/* Task 2 */

const months = ['января','февраля','марта','апреля','мая','июня','июля','августа','сентября','октября','ноября','декабря'];
/* Put your code here */
function monthNumber(s) {
  for (let i=0; i<months.length; i++) {
    if (months[i]==s) {
      return i+1;
    }
  }
}

function getDayToNewYear(d,m,y) {
  
  let inputDate = new Date(y,m,d);
  let newYear = new Date(y,12,31);
  
  return (newYear-inputDate)/(24*60*60*1000);

}

function main() {

  const d = +process.argv[2];
  const m = monthNumber(process.argv[3]);
  const y = +process.argv[4];
  
  process.stdout.write(String(getDayToNewYear(d,m,y)));
  
}

main();

File: task2.rb
# Task 2

def minRotation(str1, str2)

  return -1 if str1.size != str2.size
  
  i = str2.index(str1[0])
  if (i == nil)
    return -1
  elsif (i == 0)
    if (str1.include?(str2)) 
      return 0
    else
      return -1
    end
  else
    if ((i+1) > str1.size/2.to_i)
      j = str1.size-i
    else
      j = i
    end
	puts i
    str3 = str2[i,str2.size] + str2[0,i];
    puts str3
    if (str1 == str3)
      return j
    else
      return -1
    end
  end
end

# Put your code here
str1 = ARGV[0].to_s
str2 = ARGV[1].to_s

puts minRotation(str1, str2)



